internal/helm/helm.go

```diff
@@ -197,6 +203,33 @@ func newChartFromDirectory(directory string) (*chart.Chart, error) {
 	return loader.LoadDir(directory)
 }
 
+func newChartFromRemote(ciliumVersion semver2.Version) (*chart.Chart, error) {
+	pull := action.NewPullWithOpts(action.WithConfig(new(action.Configuration)))
+	pull.Settings = settings
+	pull.RepoURL = ciliumChart
+	pull.Version = ciliumVersion.String()
+
+	tempDir, err := os.MkdirTemp("", "cilium-cli")
+	if err != nil {
+		return nil, err
+	}
+	pull.DestDir = tempDir
+
+	_, err = pull.Run("cilium")
+	if err != nil {
+		return nil, err
+	}
+	file := path.Join(pull.DestDir, fmt.Sprintf("cilium-%s.tgz", ciliumVersion))
+	defer func() {
+		_ = os.Remove(file)
+	}()
+	helmTgz, err := os.ReadFile(file)
+	if err != nil {
+		return nil, err
+	}
+	return loader.LoadArchive(bytes.NewReader(helmTgz))
```

Comment:
Since [loader.LoadArchive](https://github.com/sayboras/cilium-cli/blob/2b9f0ead4fe18825429081e75659064593dd988b/vendor/helm.sh/helm/v3/pkg/chart/loader/archive.go#L189) needs a `io.Reader`, do we need to read the whole file at once here?

```suggestion
	fd, err := os.Open(file)
	if err != nil {
		return nil, err
	}
	defer fd.Close()
	return loader.LoadArchive(fd)
```

Reply:
Good point, I need to admit that I copied this piece from another loader 🥲